commit 0fa2c5f954c41e870fe327907c01cb8f7ea8d5a2
Author: Hans Verkuil <hans.verkuil@cisco.com>
Date:   Thu Jan 25 06:57:45 2018 -0500

    media: vivid: fix module load error when enabling fb and no_error_inj=1
    
    If the framebuffer is enabled and error injection is disabled, then
    creating the controls for the video output device would fail with an
    error.
    
    This is because the Clear Framebuffer control uses the 'vivid control
    class' and that control class isn't added if error injection is disabled.
    
    In addition, this control was added to e.g. vbi devices as well, which
    makes no sense.
    
    Move this control to its own control handler and handle it correctly.
    
    Signed-off-by: Hans Verkuil <hans.verkuil@cisco.com>
    Acked-by: Sakari Ailus <sakari.ailus@linux.intel.com>
    Cc: <stable@vger.kernel.org>      # for v4.15 and up
    Signed-off-by: Mauro Carvalho Chehab <mchehab@s-opensource.com>

diff --git a/drivers/media/platform/vivid/vivid-core.h b/drivers/media/platform/vivid/vivid-core.h
index 50802e650750..c90e4a0ab94e 100644
--- a/drivers/media/platform/vivid/vivid-core.h
+++ b/drivers/media/platform/vivid/vivid-core.h
@@ -154,6 +154,7 @@ struct vivid_dev {
 	struct v4l2_ctrl_handler	ctrl_hdl_streaming;
 	struct v4l2_ctrl_handler	ctrl_hdl_sdtv_cap;
 	struct v4l2_ctrl_handler	ctrl_hdl_loop_cap;
+	struct v4l2_ctrl_handler	ctrl_hdl_fb;
 	struct video_device		vid_cap_dev;
 	struct v4l2_ctrl_handler	ctrl_hdl_vid_cap;
 	struct video_device		vid_out_dev;